rudder-inventory-ldap/SOURCES/rudder-slapd-configure

```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+
+# Configure maxsize in /opt/rudder/etc/openldap/slapd.conf based on
+# content of /etc/default/rudder-slapd, or auto-computed based on available memory
+
+# Script specific
+PROG_NAME='rudder-slapd'
+OS=`uname -s`   # To adapt message printing
+
+
+#====================================================================
+# Message function
+#====================================================================
+message() {
+  # $1: syslog level
+  # $2: message
+
+  if [ $OS = "Linux" ]
+  then
+    logger -p "local4.$1" -s -t $PROG_NAME -i "$2"
+  else
+    # Try without option -s
+    logger -p "local4.$1" -t $PROG_NAME -i "$2"
+    echo "$PROG_NAME: $2"
+  fi
+}
+
+#====================================================================
+
+
+
+#====================================================================
+# Load specific parameters
+#====================================================================
+. /opt/rudder/etc/rudder-slapd.conf
+
+if [ -f /etc/default/$PROG_NAME ]
+then
+  . /etc/default/$PROG_NAME
+  message "info" "[INFO] Using /etc/default/$PROG_NAME for configuration"
+fi
+
+# configuration of mdbsize
+if [ "${RUDDER_MDBSIZE}" != "noauto" ]
+then
+  if [ "${RUDDER_MDBSIZE}" = "auto" ] || [ "${RUDDER_MDBSIZE}" = "" ]
+  then
+    # calculate the value to match about 50% of RAM (this is only virtual memory)
+    MEMSIZE=$(sed -n '/^MemTotal/s/[^0-9]//gp' /proc/meminfo)
+    MDBSIZE=$((${MEMSIZE} * 1024 / 2 ))
```

Comment:
MDBSIZE is the total size of the underlying database - it should not depend on RAM, but on the number of nodes or a similar metric. Or just be huge so that there is "no limit".